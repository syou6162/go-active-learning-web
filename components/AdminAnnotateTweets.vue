<template>
  <div>
    <div
      v-for="tweet in tweets"
      :key="tweet.IdStr"
      :tweet="tweet"
      :isAdmin="isAdmin"
    >
      <b-card no-body>
        <b-card-body
          :title="example | getTitle(75, '...')"
          title-tag="h5"
          class="m-1 p-2"
        >
          <img
            v-if="example.OgImage"
            v-lazy="example.OgImage"
            class="img-thumbnail img-responsive ogimage"
            onerror="this.style.display='none'"
          >
          <img
            v-if="tweet.ProfileImageUrl"
            v-lazy="tweet.ProfileImageUrl"
            class="tweet-profile-img"
            onerror="this.style.display='none'"
          >
          <a :href="'https://twitter.com/' + tweet.ScreenName + '/status/' + tweet.IdStr">@{{ tweet.ScreenName }}</a>
          <p class="card-text">
            {{ tweet.FullText }}
          </p>
          Fav: {{ tweet.FavoriteCount }}, RT: {{ tweet.RetweetCount }}, Score: {{ tweet.Score.toFixed(1) }}
          <tweet-annotate-buttons
            v-if="isAdmin"
            :tweet="tweet"
          />
          <b-card-footer>
            <b-button
              :to="example | getExampleUrl"
              class="float-right"
              size="sm"
            >
              Read more
            </b-button>
            <img
              v-if="example.Favicon"
              v-lazy="example.Favicon"
              class="example-favicon-img"
              onerror="this.style.display='none'"
            >
            <a :href="example.FinalUrl">{{ example | getDomain }} {{ example | getUserName }}</a>
          </b-card-footer>
        </b-card-body>
      </b-card>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { Example, getTitle, getDomain, getUserName, getExampleUrl } from '~/models/Example'
import TweetAnnotateButtons from '~/components/TweetAnnotateButtons.vue'
import Tweet from '~/models/Tweet'

@Component({
  components: {
    TweetAnnotateButtons,
  },
  filters: {
    getTitle(example: Example, length: number, omission: string): string {
      return getTitle(example, length, omission)
    },
    getDomain(example: Example): string {
      return getDomain(example)
    },
    getUserName(example: Example): string {
      return getUserName(example)
    },
    getExampleUrl(example: Example): string {
      return getExampleUrl(example)
    },
  }  
})

export default class AdminAnnotateTweets extends Vue {
  modalShow: boolean = false

  @Prop({required: true})
  example!: Example 
  
  @Prop({required: true})
  tweets!: Tweet[]

  @Prop({required: true, default: false})
  isAdmin!: boolean
}
</script>

<style scoped>
.example-favicon-img {
  width: 16px;
  height: 16px;
}
.tweet-profile-img {
  width: 24px;
  height: 24px;
  margin: 2px;
}
.ogimage {
  width: 128px;
  height: 96px;
  margin: 3px;
  float: right;
}
</style>
