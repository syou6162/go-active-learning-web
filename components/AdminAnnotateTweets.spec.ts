import { createLocalVue, mount } from '@vue/test-utils'
import { ButtonPlugin, BCard, BCardBody, BCardFooter } from 'bootstrap-vue'
import AdminAnnotateTweets from './AdminAnnotateTweets.vue'
import VueLazyLoad from 'vue-lazyload'

const localVue = createLocalVue();
localVue.use(VueLazyLoad);
localVue.use(ButtonPlugin)
localVue.component('b-card', BCard)
localVue.component('b-card-body', BCardBody)
localVue.component('b-card-footer', BCardFooter)

const tweets = [
  {
    ExampleId: 21805423,

    CreatedAt: "2020-08-26T03:13:06Z",
    IdStr: "1298458422581448704",
    FullText: "【イベント情報】先月開催された「Embulk &amp; DigDag Online Meetup 2020」のレポート記事を公開しました。\nEmbulkとDigdagをプロダクション環境で利用している事例や、Embulk開発者による発表が盛り沢山です！是非ご一読下さい。\nhttps://t.co/gdtRuoyJtY",
    FavoriteCount: 6,
    RetweetCount: 4,
    Lang: "ja",

    ScreenName: "trocco_jp",
    Name: "データ分析基盤の総合支援SaaS「trocco(トロッコ)」",
    ProfileImageUrl: "https://pbs.twimg.com/profile_images/1051692644311556096/WmDcCj4Z_normal.jpg",
    Label: 1,
    Score: 0.4894515842007917
  }
]

describe('AdminAnnotateTweets', () => {
  test('管理者用のtweetアノテーション画面が表示される', () => {
    const wrapper = mount(AdminAnnotateTweets, {
      localVue,
      propsData: {
        example: {
          Id: 11111,
          Label: 0,
          Url: "https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020",
          FinalUrl: "https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020",
          Title: "Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)",
          Description: "",
        },
        tweets: tweets,
        isAdmin: true
      },
    })
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
