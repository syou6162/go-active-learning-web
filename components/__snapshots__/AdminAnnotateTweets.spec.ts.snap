// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminAnnotateTweets 管理者用のtweetアノテーション画面が表示される 1`] = `
<div>
  <div
    isadmin="true"
    tweet="[object Object]"
  >
    <div
      class="card"
    >
      <!---->
      <!---->
      <div
        class="card-body m-1 p-2"
      >
        <h5
          class="card-title"
        >
          Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
        </h5>
        <!---->
        <!---->
         
        <img
          class="tweet-profile-img"
          onerror="this.style.display='none'"
        />
         
        <a
          href="https://twitter.com/trocco_jp/status/1298458422581448704"
        >
          @trocco_jp
        </a>
         
        <p
          class="card-text"
        >
          
          【イベント情報】先月開催された「Embulk &amp; DigDag Online Meetup 2020」のレポート記事を公開しました。
EmbulkとDigdagをプロダクション環境で利用している事例や、Embulk開発者による発表が盛り沢山です！是非ご一読下さい。
https://t.co/gdtRuoyJtY
        
        </p>
        
        Fav: 6, RT: 4, Score: 0.5
        
        <div>
          
  Annotate as: 
  
          <button
            aria-pressed="false"
            autocomplete="off"
            class="btn btn-primary btn-sm"
            type="button"
          >
            
    Positive
  
          </button>
           
          <button
            class="btn btn-outline-danger btn-sm"
            type="button"
          >
            
    Negative
  
          </button>
           
          <button
            class="btn btn-outline-secondary btn-sm"
            type="button"
          >
            
    Unlabeled
  
          </button>
        </div>
         
        <div
          class="card-footer"
        >
          <a
            class="btn float-right btn-secondary btn-sm"
            href="/example/11111"
            target="_self"
          >
            
            Read more
          
          </a>
           
          <!---->
           
          <a
            href="https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
          >
            blog.trocco.io 
          </a>
        </div>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
